### PR TITLE
REG-518: NewHeaderBlock sandbox event, sandbox network config merging

### PR DESCRIPTION
**Pull request by nickshatilo**
Added `NewHeaderBlock` event emitting to `MockCheckpointManager`. Improved `sandbox:rebuild` so that it now updates `uns-config.json`

### CODE REVIEW

File: test/RootRegistry.test.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ const {
 } = require('./helpers/polygon');
 
 const { utils } = ethers;
+const { bufferToHex } = require('ethereumjs-util');
```
[the remark this answers is not in the document]
Comment (nickshatilo) [reply test/RootRegistry.test.js:15]: Absolutely agree, missed this one. Thank you

File: contracts/@maticnetwork/pos-portal/MockCheckpointManager.sol [hunk 2]
```diff
@@ -3,6 +3,15 @@
...
 contract ICheckpointManager {
+    event NewHeaderBlock(
+        address indexed proposer,
+        uint256 indexed headerBlockId,
+        uint256 indexed reward,
+        uint256 start,
+        uint256 end,
+        bytes32 root
+    );
+
```
[the remark this answers is not in the document]
Comment (nickshatilo) [reply contracts/@maticnetwork/pos-portal/MockCheckpointManager.sol:14]: Ah, yeah. Wasn't thinking this way. Great catch

File: test/RootRegistry.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickshatilo) [reply test/RootRegistry.test.js]: Yeah, so I cleaned up `test/helpers/polygon.js` to just use ethers.js :)

**Review by aquiladev — APPROVED** [review]
Looks good. Do not forget to increase patch version and create a tag for the new version

**Pull request merged**